src/configurePipeline/services/target/azureTargetService.ts

```diff
@@ -0,0 +1,35 @@
+import { AzureTreeDataProvider } from "vscode-azureextensionui";
+
+import { AzureTargetResource, PipelineTargets } from "../../model/Common";
+import { WebAppTreeItem } from "../../tree/WebAppTreeItem";
+
+export class AzureTargetService {
+    private targetResource: AzureTargetResource;
+    private azureTreeDataProvider: AzureTreeDataProvider;
+
+    constructor(azureTreeDataProvider: AzureTreeDataProvider) {
+        this.azureTreeDataProvider = azureTreeDataProvider;
+    }
+
+    public async getTargetResource(): Promise<AzureTargetResource> {
+        let azureNode = <WebAppTreeItem>await this.azureTreeDataProvider.showTreeItemPicker("appService");
+        return this.extractTargetFromNode(azureNode);
+    }
+
+    public extractTargetFromNode(azureNode: WebAppTreeItem) {
```

Comment:
return type here. Also, should this name be consistent with the function name above like getTargetResourceFromNode ?